src/ring_theory/adjoin_root.lean

```diff
@@ -126,9 +126,8 @@ instance (α : Type*) [comm_semiring α] {K : Type u} [field K]
 instance [comm_semiring S] [algebra S R] : algebra S (adjoin_root f) :=
 ideal.quotient.algebra S
 
-instance [comm_semiring S] [comm_semiring K] [has_smul S K] [algebra S R] [algebra K R]
-  [is_scalar_tower S K R] :
-  is_scalar_tower S K (adjoin_root f) :=
+instance is_scalar_tower' [comm_semiring S] [comm_semiring K] [has_smul S K]
```

Comment:
@eric-wieser, are these two overlapping instances of `is_scalar_tower` & an action on `adjoin_root` likely to cause issues?

Reply:
Overlapping instances of `is_scalar_tower` should be no problem, since the class lives in `Prop` all instances are defeq anyway. Overlapping actions might be more problematic, which ones are you referring to? (I only see the `distrib_mul_action` which is compatible with the `has_smul`.)

Reply:
I was talking about the `algebra` and the `distrib_smul`, but now I see that they're both inherited from `submodule/ideal.quotient` so they should both be fine

Reply:
I'm merging them, there's a more general statement that covers both.